src/node_pointer.h

```diff
@@ -5,62 +5,13 @@
 
 #include <nan.h>
 
-/*
- * Called when the "pointer" is garbage collected.
- */
-
-// UNUSED: however, inline functions don't generate code.
-inline static void wrap_pointer_cb(char *data, void *hint) {
-  // fprintf(stderr, "wrap_pointer_cb\n");
-}
-
-/*
- * Wraps "ptr" into a new SlowBuffer instance with size "length".
- */
-
-inline static v8::Local<v8::Value> WrapPointer(void *ptr, size_t length) {
-  return Nan::NewBuffer(static_cast<char *>(ptr), length, wrap_pointer_cb, 0)
-      .ToLocalChecked();
-}
-
-/*
- * Wraps "ptr" into a new SlowBuffer instance with length 0.
- */
-
-inline static v8::Local<v8::Value> WrapPointer(void *ptr) {
-  return WrapPointer(ptr, 0);
-}
-
-/*
- * Unwraps Buffer instance "buffer" to a C `char *` with the offset specified.
- */
-
-inline static char *UnwrapPointer(v8::Local<v8::Value> buffer, int64_t offset) {
-  if (node::Buffer::HasInstance(buffer)) {
-    return node::Buffer::Data(buffer.As<v8::Object>()) + offset;
-  } else {
-    return 0;
-  }
-}
-
-/*
- * Unwraps Buffer instance "buffer" to a C `char *` (no offset applied).
- */
-
-inline static char *UnwrapPointer(v8::Local<v8::Value> buffer) {
-  if (node::Buffer::HasInstance(buffer)) {
-    return node::Buffer::Data(buffer.As<v8::Object>());
-  } else {
-    return 0;
-  }
-}
-
-/**
- * Templated version of UnwrapPointer that does a reinterpret_cast() on the
- * pointer before returning it.
- */
-
-template <typename Type>
-inline static Type UnwrapPointer(v8::Local<v8::Value> buffer) {
-  return reinterpret_cast<Type>(UnwrapPointer(buffer));
-}
+struct nanomsg_socket_s {
+  uv_poll_t poll_handle;
+  uv_os_sock_t sockfd;
+  Nan::Callback *callback;
+
+  uv_mutex_t close_mutex;
+  uv_mutex_t free_mutex;
+  bool close_called;
+  bool free_called;
```

Comment:
please change the name to `nanomsg_poll_ctx` and refer to it throughout the code as `struct nanomsg_poll_ctx`.